MigrationPreparer/MigrationPreparerImplementation.h

```diff
@@ -0,0 +1,157 @@
+/*
+* If not stated otherwise in this file or this component's LICENSE file the
+* following copyright and licenses apply:
+*
+* Copyright 2024 RDK Management
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#pragma once
+
+#include "Module.h"
+#include <interfaces/IMigrationPreparer.h>
+#include "tracing/Logging.h"
+#include <fstream>
+#include <string>
+#include <map>
+#include <mutex>
+#include <vector>
+#include <cerrno>
+#include <unordered_set>
+#include <list>
+#include "secure_wrapper.h"
+#include "UtilsJsonRpc.h"
+#include "Module.h"
+
+
+#define MIGRATIONPREPARER_NAMESPACE "MigrationPreparer"
+#define DATASTORE_PATH _T("/opt/secure/migration/migration_data_store.json")
+#define DATASTORE_DIR _T("/opt/secure/migration/")
+#define TR181_MIGRATION_READY "Device.DeviceInfo.X_RDKCENTRAL-COM_RFC.Bootstrap.MigrationReady"
+#define MIGRATIONREADY_PATH _T("/opt/secure/migration/migrationready.txt")
+#define MIGRATIONREADY_DIR _T("/opt/secure/migration/")
+
+// PLUGIN SPECIFIC CUSTOM ERROR CODES
+#define ERROR_CREATE    4
```

Comment:
Yes, the error codes can be found in 2 separate files namely MigrationPreparerImplementation.h and MigrationPreparer.h.
MigrationPreparerImplementation.cpp is compiled to form libWPEFrameworkMigrationPreparerImplementation.so
MigrationPreparer.cpp and MigrationPreparerJsonRpc.cpp is compiled to form libWPEFrameworkMigrationPreparer.so.
Since above .so file's don't share any common file other than Module.h, in order to make these .so's aware of plugin specific error codes, error macros were added to both of the .so specific header files.